carts/views.py

```diff
@@ -0,0 +1,62 @@
+import json
+
+from django.http import JsonResponse
+from django.views import View
+from django.db.models import Q
+
+from carts.models import Cart
+from users.models import User
+from products.models import Product
+from users.decorator import login_decorator
+
+class CartView(View):
+    @login_decorator
+    def post(self, request):
+        try:
+            data       = json.loads(request.body)
+            product_id = data['product_id']
+            quantity   = data['quantity']
+
+            if not Product.objects.filter(id=product_id).exists():
+                return JsonResponse({"MESSAGE":"DOES_NOT_EXIST_ERROR"}, status=400)
+
+            if Cart.objects.filter(product_id=product_id).exists():
+                return JsonResponse({"MESSAGE":"ALREADY_EXIST"}, status=400)
+
+            Cart.objects.create(
+                user       = request.user,
+                product_id = product_id,
+                quantity   = quantity,
+                )
+
+            return JsonResponse({'MESSAGE':'CREATE'}, status=201)
+
+        except KeyError:
+            return JsonResponse({'MESSAGE':'KEY_ERROR'}, status=400)
+
+    @login_decorator
+    def patch(self, request):
+        try:
+            data       = json.loads(request.body)
+            product_id = data['product_id']
+            quantity   = data['product_quantity']
+            cart       = Cart.objects.filter(product_id=product_id)
+            cart.update(quantity=quantity)
+
+            return JsonResponse({'MESSAGE':'SUCCESS'}, status=201)
+
+        except KeyError:
+            return JsonResponse({'MESSAGE':'KEY_ERROR'}, status=400)
```

Comment:
1.애초에 없데이트 하려고하는 카트 ID를 받으면 코드가 훨씬 간단해 질 것 같네요!
DELETE에서 드린 피드백과 마찬가지로 REST API에서 UPDATE 할 때, 자원 하나를 업데이트 한다고 하면 `path paramter`를 사용해서 구현해야 합니다.

```python
PATCH /cart/1
```
2. cart를 업데이트할 때, 유저도 같이 조회해야 합니다! 그렇지 않으면, 로그인이 가능한 유저는 어떠한 카트도 업데이트할 수 있는 버그가 발생할 수 있습니다.

```suggestion
    @login_decorator
    def patch(self, request, id):
        try:
            data       = json.loads(request.body)
            quantity   = data['product_quantity']
            cart       = Cart.objects.filter(id=cart_id, user_id = ...)
            cart.update(quantity=quantity)

            return JsonResponse({'MESSAGE':'SUCCESS'}, status=201)

        except KeyError:
            return JsonResponse({'MESSAGE':'KEY_ERROR'}, status=400)
```

Reply:
cart id 전달받아 수정하는 방식으로 수정하였고, user_id에 의한 필터도 추가하였습니다.